packages/components/src/components/Warning/Warning.tsx

```diff
@@ -158,7 +165,29 @@ export const Warning = ({
                     })}
                 />
             )}
-            {children}
+            <Row justifyContent="space-between" gap={spacings.lg} flex={1}>
+                {children}
+                {rightContent && (
+                    <WarningContext.Provider value={{ variant }}>
+                        {rightContent}
+                    </WarningContext.Provider>
+                )}
+            </Row>
         </Wrapper>
     );
 };
+
+const WarningButton = ({ children, ...rest }: ButtonProps) => {
```

Comment:
Shouldn't this be in its own file?

Reply:
It makes sense, I separated it to multiple files.